### ISSUE TITLE
wasm-pack + serde + AtomicUsize 

### ISSUE THREAD

**Issue by gakonst** [serde-rs/serde#1778]
`Serialize` is not implemented for `AtomicUsize` in wasm when using [`wasm-pack`](https://github.com/rustwasm/wasm-pack).

Example:

1. Install `wasm-pack`: `curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh`

2. Add to your Cargo.toml:
```toml
[lib]
crate-type = ["cdylib", "lib"]

[dependencies]
serde = { version = "1.0.106", features = ["derive"] }
wasm-bindgen = "0.2.60"
```

3. Code:
```rust
use std::sync::atomic::AtomicUsize;
use serde::Serialize;

#[derive(Serialize)]
pub struct S(AtomicUsize);
```
4. Run `wasm-pack build` (`wasm-pack` builds into `wasm32-unknown-unknown`)

It errors with:
```
pub struct S(AtomicUsize);
  |              ^^^^^^^^^^^ the trait `_IMPL_SERIALIZE_FOR_S::_serde::Serialize` is not implemented for `std::sync::atomic::AtomicUsize
```

Could anybody explain to me why/if it is not fundamentally impossible to implement this?

**Comment by dtolnay** [comment]
This looks like a consequence of https://github.com/serde-rs/serde/pull/1581. Determining exactly which targets can handle which atomic operations is complicated so we've only started with a conservative set and would need to expand as use cases come up.

**Comment by gakonst** [comment]
@dtolnay thank you for the quick response, your approach makes sense. It _seems_ like `wasm32-unknown-unknown` has support for Atomics since https://github.com/rust-lang/rust/pull/54017?

**Comment by dtolnay** [comment]
Sounds good; someone should send a PR to enable the atomic impls on wasm32-unknown-unknown on the correct range of compiler versions.

**Comment by dtolnay** [comment]
Fixed in serde 1.0.150 by #2337.
